### PR TITLE
Bump several dependencies to address some known CVEs

### PR DESCRIPTION
**Pull request by zyga**
This is a far cry from the overall situation, as there are plenty of dependencies that are still not up-to-date, but this set is, while conservative, still useful and, most importantly, compatible.

The change passes all tests and also seems to work in practice on my local installation.

### CODE REVIEW

**Comment by stefbehl** [comment]
Thanks @zyga for contributing this PR. We'll take a look.

**Comment by zyga** [comment]
Thank you @stefbehl, please be gentle with me as I'm not a Java developer and I'm learning my way around still.

Do you have plans to move to more recent versions of spring and vaadin?

**Review by stefbehl — CHANGES_REQUESTED** [review]
Hi @zyga, thanks again for this PR!

In the POM, you also need to update the parent version:
` <parent>
      <groupId>org.springframework.boot</groupId>
      <artifactId>spring-boot-starter-parent</artifactId>
      <version>2.3.12.RELEASE</version>
   </parent>` 

Did you check the compatibility of Spring Boot 2.7.8 and Spring Cloud?
-> https://spring.io/projects/spring-cloud
I'm pretty sure you also need to update Spring Cloud.

Best regards.

**Comment by zyga** [comment]
Thank you, for the review.

I have not checked compatibility with spring-cloud, merely ran tests and deployed that to my toy instance I use for various interactions.

**Comment by zyga** [comment]
With two more bumps:

``` diff
diff --git a/pom.xml b/pom.xml
index e029c0113..09cd57b92 100644
--- a/pom.xml
+++ b/pom.xml
@@ -132,7 +132,7 @@
 
       <spring.boot.version>2.7.8</spring.boot.version>
       <spring-framework.version>5.3.25</spring-framework.version>
-      <spring.cloud.version>Hoxton.SR12</spring.cloud.version>
+      <spring.cloud.version>2021.0.5</spring.cloud.version>
       <spring.plugin.core.version>2.0.0.RELEASE</spring.plugin.core.version>
 
       <snapshotDependencyAllowed>true</snapshotDependencyAllowed>

```

``` diff
diff --git a/pom.xml b/pom.xml
index 6b866f8e9..e029c0113 100644
--- a/pom.xml
+++ b/pom.xml
@@ -16,7 +16,7 @@
    <parent>
       <groupId>org.springframework.boot</groupId>
       <artifactId>spring-boot-starter-parent</artifactId>
-      <version>2.3.12.RELEASE</version>
+      <version>2.7.8</version>
    </parent>
 
    <groupId>org.eclipse.hawkbit</groupId>
```

I get the following build failure:

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.10.1:compile (default-compile) on project hawkbit-repository-core: Compilation failure
[ERROR] /home/zyga/hawkbit/hawkbit-repository/hawkbit-repository-core/src/main/java/org/eclipse/hawkbit/repository/model/helper/EventPublisherHolder.java:[56,55] cannot find symbol
[ERROR]   symbol:   method getServiceId()
[ERROR]   location: variable serviceMatcher of type org.springframework.cloud.bus.ServiceMatcher
```

**Comment by zyga** [comment]
Looking at https://github.com/spring-cloud/spring-cloud-bus/pull/239/files it seems this has been replaced by `getBusId()` now.

The build now fails with:

```
[ERROR] warning: unknown enum constant javax.annotation.meta.When.MAYBE
[ERROR]   reason: class file for javax.annotation.meta.When not found
[ERROR] Note: Hibernate JPA 2 Static-Metamodel Generator 5.4.32.Final
[ERROR] warning: unknown enum constant javax.annotation.meta.When.MAYBE
[ERROR]   reason: class file for javax.annotation.meta.When not found
[ERROR] /home/zyga/hawkbit/hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/RepositoryApplicationConfiguration.java:112: error: package org.hibernate.validator does not exist
[ERROR] import org.hibernate.validator.HibernateValidatorConfiguration;
[ERROR]                               ^
[ERROR] warning: unknown enum constant javax.annotation.meta.When.MAYBE
[ERROR]   reason: class file for javax.annotation.meta.When not found
```

**Comment by MartB** [comment]
I ported this already but they dont pass all tests, i can make my changes available later!

**Comment by stefbehl** [comment]
Good morning @zyga and @MartB, thanks for actively driving the Spring Boot migration! FYI, the Hawkbit committer team is currently actively working on the new milestone release M8 at the moment and we aim at including the Spring Boot 2.7.8 migration to get rid of the CVEs (depending on how disruptive it will be). The migration to Spring Boot 3.* will be a separate step after M8. I'll keep you posted.

**Comment by zyga** [comment]
Given the extent of changes proposed by @MartB I have a feeling this is above my level of familiarity with Spring and related libraries.

Does it make sense to keep some of the changes in this branch or would it make sense to just drop and let the people that know more about it drive the work?

**Comment by stefbehl** [comment]
@zyga @MartB We can offer to take the boot migration over in the context of the M8 release activities.
I consolidated the changes from both of you in a feature branch and added some further changes to fix the failing tests.
See https://github.com/bosch-io/hawkbit/tree/feature/spring-boot-27-migration
We decided to temporarily downgrade H2 to prioritize the MySql setup (but we'll revisit H2 soon).
I'll keep you posted. Regards Stefan

**Comment by MartB** [comment]
Thanks @stefbehl and @zyga!

As im currently deploying hawkbit as part of an university project, an official M8 release with boot 2.7 engineered by people more familiar with the codebase would be terrific.
Will monitor the progress and test/review if needed!

**Comment by stefbehl** [comment]
We just opened a PR for the Spring Boot 2.7.8 migration: https://github.com/eclipse/hawkbit/pull/1320
Feel free to review the PR and leave your comments! Thanks!

All of the tests are passing now. So far we've been focusing on the MySQL 8 setup. We decided to downgrade H2 temporarily, as it is not really meant for production scenarios. We'll revisit the H2 setup soon.

**Comment by MartB** [comment]
> Thank you @stefbehl, please be gentle with me as I'm not a Java developer and I'm learning my way around still.
> 
> Do you have plans to move to more recent versions of spring and vaadin?

Regarding Vaadin: https://github.com/eclipse/hawkbit/pull/1320#discussion_r1116054826

I am absolutely not familiar with Vaadin, but if someone is willing to pick this up, it would get us out of this deadlock.

**Comment by stefbehl** [comment]
@zyga I think we can close this PR, cant' we? We are currently finalizing the [Spring Boot migration](https://github.com/eclipse/hawkbit/pull/1320). Once the Spring Boot migration PR is merged, we will builds the milestone release.

**Comment by zyga** [comment]
@stefbehl yes, thank you for picking up that work :)

**Pull request closed without merging**
